crates/router/src/core/payment_methods/cards.rs

```diff
@@ -111,30 +134,44 @@ pub async fn add_payment_method(
     req.validate()?;
     let merchant_id = &merchant_account.merchant_id;
     let customer_id = req.customer_id.clone().get_required_value("customer_id")?;
-    let response = match req.card.clone() {
-        Some(card) => {
-            add_card_to_locker(&state, req.clone(), &card, &customer_id, merchant_account)
-                .await
-                .change_context(errors::ApiErrorResponse::InternalServerError)
-                .attach_printable("Add Card Failed")
-        }
-        None => {
-            let pm_id = generate_id(consts::ID_LENGTH, "pm");
-            let payment_method_response = api::PaymentMethodResponse {
-                merchant_id: merchant_id.to_string(),
-                customer_id: Some(customer_id.clone()),
-                payment_method_id: pm_id,
-                payment_method: req.payment_method,
-                payment_method_type: req.payment_method_type,
-                card: None,
-                metadata: req.metadata.clone(),
-                created: Some(common_utils::date_time::now()),
-                recurring_enabled: false,           //[#219]
-                installment_payment_enabled: false, //[#219]
-                payment_experience: Some(vec![api_models::enums::PaymentExperience::RedirectToUrl]), //[#219]
-            };
-            Ok((payment_method_response, false))
-        }
+
+    let response = match req.payment_method {
+        api_enums::PaymentMethod::BankTransfer => match req.bank_transfer.clone() {
+            Some(bank) => add_bank_to_locker(
```

Comment:
@NishantJoshi00 @Chethan-rao @ShankarSinghC It's better if we have your review here for locker-related stuff.

Reply:
@NishantJoshi00 @Chethan-rao @ShankarSinghC - can you guys add your review for the locker bit?